src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -141,13 +134,20 @@ module.exports = function<T, P, I, TI, C, CX>(config : HostConfig<T, P, I, TI, C
       const root : FiberRoot = (container.stateNode : any);
       const current = root.current;
 
-      root.pendingContext = getContextForSubtree(parentComponent);
+      const context = getContextForSubtree(parentComponent);
+      if (root.context === null) {
```

Comment:
This seems gross. If I always set pendingContext instead, this test fails, because hasContextChanged() returns true and we rerender Middle despite element equality:

https://github.com/facebook/react/blob/c79af45f1e1bb9fc0ef9af27b586523832c5f6b8/src/renderers/shared/fiber/__tests__/ReactIncremental-test.js#L338

This is because the root hasn't completed when we do the update and so .pendingContext hasn't been moved to .context yet. (?) I think I'm missing something here.